src/library/scala/collection/Iterable.scala

```diff
@@ -228,6 +228,88 @@ trait IterableOps[+A, +CC[_], +C] extends Any with IterableOnce[A] with Iterable
   /** A view over the elements of this collection. */
   def view: View[A] = View.fromIteratorProvider(() => iterator)
 
+  /** Compares the size of this $coll to a test value.
+    *
+    *   @param   _size the test value that gets compared with the size.
+    *   @return  A value `x` where
+    *   {{{
+    *        x <  0       if this.size <  _size
+    *        x == 0       if this.size == _size
+    *        x >  0       if this.size >  _size
+    *   }}}
+    *  The method as implemented here does not call `size` directly; its running time
+    *  is `O(size min _size)` instead of `O(size)`. The method should be overwritten
+    *  if computing `size` is cheap.
+    */
+  def sizeCompare(_size: Int): Int = {
```

Comment:
not super happy with this parameter name, but I wasn't sure what else to call it, as shadowing the `size` method would be confusing

Reply:
Maybe `otherSize` or `thatSize`?